internal/app/coroutines/completePromise.go

```diff
@@ -165,8 +165,9 @@ func CompletePromise(metadata *metadata.Metadata, req *t_api.Request, res func(*
 			} else {
 				status := t_api.ForbiddenStatus(p.State)
 				strict := req.CompletePromise.Strict && p.State != req.CompletePromise.State
+				timeout := !req.CompletePromise.Strict && req.CompletePromise.State.In(promise.Timedout)
```

Comment:
```suggestion
				timeout := !req.CompletePromise.Strict && p.State == promise.Timedout
```

We need to check the state of the promise, as opposed to the state in the request

Reply:
This makes sense, thanks for clarifying that timeout must not be allowed on request.